doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -185,129 +166,118 @@ Newly opened modules and sections inherit the current settings.
+.. cmd:: Print Assumptions @smart_qualid
 
-      Displays the set of transparent constants :n:`@qualid` relies on
-      in addition to the assumptions.
+   Displays all the assumptions (axioms, parameters and
+   variables) a theorem or definition depends on.
 
-   .. cmdv:: Print All Dependencies @qualid
-      :name: Print All Dependencies
+   .. exn:: Closed under the global context
 
-      Displays all assumptions and constants :n:`@qualid` relies on.
+      The theorem or definition doesn't depend on any assumptions.
 
+.. cmd:: Print Opaque Dependencies @smart_qualid
 
-.. cmd:: Search @qualid
+   Displays the assumptions and opaque constants that :n:`@smart_qualid` depends on.
 
-   This command displays the name and type of all objects (hypothesis of
-   the current goal, theorems, axioms, etc) of the current context whose
-   statement contains :n:`@qualid`. This command is useful to remind the user
-   of the name of library lemmas.
+.. cmd:: Print Transparent Dependencies @smart_qualid
 
-   .. exn:: The reference @qualid was not found in the current environment.
+   Displays the assumptions and  transparent constants that :n:`@smart_qualid` depends on.
 
-      There is no constant in the environment named qualid.
+.. cmd:: Print All Dependencies @smart_qualid
 
-   .. cmdv:: Search @string
+   Displays all the assumptions and constants :n:`@smart_qualid` relies on.
 
-      If :n:`@string` is a valid identifier, this command
-      displays the name and type of all objects (theorems, axioms, etc) of
-      the current context whose name contains string. If string is a
-      notation’s string denoting some reference :n:`@qualid` (referred to by its
-      main symbol as in `"+"` or by its notation’s string as in `"_ + _"` or
-      `"_ 'U' _"`, see Section :ref:`notations`), the command works like ``Search`` :n:`@qualid`.
+.. todo: move Search* to the section on query commands once we agree on the wording
 
-   .. cmdv:: Search @string%@ident
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      The string string must be a notation or the main
-      symbol of a notation which is then interpreted in the scope bound to
-      the delimiting key :token:`ident` (see Section :ref:`LocalInterpretationRulesForNotations`).
+   .. insertprodn search_item search_item
 
-   .. cmdv:: Search @term_pattern
+   .. prodn::
+      search_item ::= {? - } @string {? @scope }
+      | {? - } @one_term
 
-      This searches for all statements or types of
-      definition that contains a subterm that matches the pattern
-      :token:`term_pattern` (holes of the pattern are either denoted by `_` or by
-      :n:`?@ident` when non linear patterns are expected).
+   Todo: :n:`@scope ::= % @ident` (fix in syntax extensions)
 
-   .. cmdv:: Search {+ {? -}@term_pattern_string}
+   Displays the name and type of objects matching :n:`@search_item`\s
+   in the selected goals (if a proof is open)
```

Comment:
```suggestion
   in the context of the selected goal (if a proof is open)
```

Or even better, use a formulation closer to the one you used for `SearchHead`, `SearchPattern` and `SearchRewrite` (very good!).